applications/StructuralMechanicsApplication/structural_mechanics_application.cpp

```diff
@@ -52,6 +52,9 @@ KratosStructuralMechanicsApplication::KratosStructuralMechanicsApplication():
     mShellThinCorotationalElement3D3N( 0, Element::GeometryType::Pointer( new Triangle3D3 <Node<3> >( Element::GeometryType::PointsArrayType( 3 ) ) ), true ),
     // Adding the membrane element
     mMembraneElement3D3N( 0, Element::GeometryType::Pointer( new Triangle3D3 <Node<3> >( Element::GeometryType::PointsArrayType( 3 ) ) ) ),
+	mMembraneElement3D4N( 0, Element::GeometryType::Pointer( new Quadrilateral3D4 <Node<3> >( Element::GeometryType::PointsArrayType( 4 ) ) ), false ),	
```

Comment:
I have checked the existing membrane element at the beginnning before I start my implementation and I found out there is no pre-stress in the element. And I was not able to figure out the theory behind it based on the code although I had really tried it...
I haved checked the existing membrane element now again. I would say the difference might lie in the different mathematical derivation of the same theory (the new implementation is based on green-lagrange strain, I guess it is in the case of the existing membrane element). This might lead to differences in performance. It would be good, if you can give me some more details about the existing membrane element.
The reason for using new implementation is that I knew at the beginning that the new implementation will work and it is not a very trivial task. And the new implementation is possibly more code efficient when considering various form finding functions adding to the basic element formulation.

It is of course meaningful at the end to try to adapt the form finding to the existing element and then to comparing the two elements (with and without form finding).

Best regards,
Long





Regarding your comments I think it is meaningful to compare the both implementation
Can you give me some sources regarding to the existing element.